packages/angular/src/schematics/utils/config.ts

```diff
@@ -88,11 +92,48 @@ export function addArchitectBuilder(
   builderOpts: any
 ): void | never {
   const config = readConfig(host);
-  const appConfig = getAngularAppConfig(config, projectName);
-  appConfig.architect[builderName] = builderOpts;
+  const angularJson = getAngularJson(config, projectName);
+  angularJson.architect[builderName] = builderOpts;
   writeConfig(host, config);
 }
 
+/**
+ * Updates the angular.json to add an additional schematic collection
+ * to the CLI configuration.
+ */
+export function addCli(host: Tree, collectionName: string): void | never {
+  const angularJson = readConfig<any>(host);
+
+  if (angularJson.cli === undefined) {
+    angularJson.cli = {};
+  }
+
+  if (angularJson.cli.schematicCollections === undefined) {
+    angularJson.cli.schematicCollections = [];
+  }
+
+  angularJson.cli.schematicCollections.push(collectionName);
+
+  writeConfig(host, angularJson);
+}
+
+// TODO can remove [property: string]: any; when upgrading @angular/cli dev-dep to v16 or later
```

Comment:
We need to have a tech debt ticket for this. You can throw it in the Ionic 8 epic since we'll be support ng16+ in that version.